openedx/features/idea/api_views.py

```diff
@@ -9,18 +9,23 @@
 
 class FavoriteAPIView(APIView):
     """
-    FavoriteAPIView is used t toggle favorite idea for the user
+    FavoriteAPIView is used to toggle favorite idea for the user
     """
     authentication_classes = (SessionAuthentication, BasicAuthentication)
     permission_classes = (IsAuthenticated,)
 
     def post(self, request, idea_id):
         response = dict(message='User is added to favorites')
-        http_status = status.HTTP_201_CREATED
+        toggle_status = status.HTTP_201_CREATED
         user = request.user
-        idea = Idea.objects.get(pk=idea_id)
-        if not idea.toggle_favorite(user):
-            response['message'] = 'User is removed from favorites'
-            http_status = status.HTTP_200_OK
+        try:
+            idea = Idea.objects.get(pk=idea_id)
```

Comment:
I think only `idea = Idea.objects.get(pk=idea_id)` should be inside `try` block otherwise if there is any issue later in the block it will throw 404 error in the response.